### PR TITLE
Replace CashCountsStatistic error calculation by analytical expression

### PR DESCRIPTION
**Pull request by registerrier**
**Description**

This pull request replaces the current asymmetric error calculation as well as the `n_sig_matching_significance` function by analytical expression for the `CashCountsStatistic`. This uses the `scipy.special.lambertw` function.
The upper limit calculation should be adapted too.

This makes the error calculation more than 1e3 times faster than the current root searching scheme.

I don't know if there is an analytical solution for the WStat. 

**Dear reviewer**

### CODE REVIEW

**Review by QRemy — APPROVED** [review]
Thanks @registerrier, looks good !

**Comment by adonath** [comment]
Nice @registerrier! This is especially useful for the  `ExcessMapEstimator`, where the root finding is run for each pixel.

File: gammapy/stats/counts_statistic.py [hunk 1]
```diff
@@ -315,6 +316,37 @@ def sum(self, axis=None):
     def __getitem__(self, key):
         return CashCountsStatistic(n_on=self.n_on[key], mu_bkg=self.n_bkg[key])
 
+    def compute_errn(self, n_sigma=1.0):
```
Comment (bkhelifi) [inline gammapy/stats/counts_statistic.py:319]: Should the docstrings for the 3 functions be repeated?

**Review by bkhelifi — COMMENTED** [review]
Thanks a lot... Except a small question about the docstring, it is ok for me!

**Review by adonath — APPROVED** [review]
Thanks @registerrier! I have nor further comments, I think this is ready to be merged?

**Comment by registerrier** [comment]
I have added the `compute_upper_limit` calculation as well and modified the tolerance of sensitivity tests in test_ excess.py.

**Pull request merged**